boilerplate/flyte/end2end/run-tests.py

```diff
@@ -98,10 +98,14 @@ def executions_finished(executions_by_wfgroup: Dict[str, List[FlyteWorkflowExecu
     return True
 
 def sync_executions(remote: FlyteRemote, executions_by_wfgroup: Dict[str, List[FlyteWorkflowExecution]]):
-    for executions in executions_by_wfgroup.values():
-        for execution in executions:
-            print(f"About to sync execution_id={execution.id.name}")
-            remote.sync(execution)
+    try:
```

Comment:
can we add a log line just so we can see how often this is happening?

Reply:
where/how?  Do you have some infrastructure to collect events, count?  

Ex: Spin up a HTTP endpoint that can write to a database.  [ I doubt anyone thoroughly goes through print statements across runs ]

Reply:
I can write something to collect logs to viz and explore the runs...  [ but then would want a GCP Project to use for logging ].  Else do advise what the Flyte OSS Project uses for monitoring things like this.

Reply:
i just mean a print statement... is that not okay?  i won't look at it most of the time, but i will some times.

Reply:
that's fine... but I am offering to do more --> write something so can actually run a query ( ex: SQL ) on counts [ # of runs, # of trys, # of sucesses, etc ].  Do we have a GCP Project that can be used to create this?  Or, that is not desired?

Reply:
Thought was to write to BigQuery -- unless there is some other convention.

Reply:
for this particular log line, i think probably just printing is okay.  for the test runs in general, yeah that would be fantastic.  why gcp in particular? aren't these hitting the aws account? also sounds like a lot to set up too.

Reply:
since I can imagine wanting to run queries to understand things -- it seems worth investing in logging [ also a general habbit :-) ].  I mean, when having permissions, creating such things is very easy, so at least doesn't seem like alot to me 🤷 .  

Hitting the AWS account ...  I don't know.  There might be built in features in AWS, but I don't know where/how I'd query that?  BigQuery is my go-to datastore for lots of things, which seems not-inappropriate given the desired query patterns.  I can just print and rather than log/store, but then we won't have the ability to query ... Otherwise, I'm offering to do it [ but need permissions, and a project to write to and that has associated billing, though the charges will be essentially if not actually nonexistent ].

Reply:
okay yeah let's table it for now then.  print is fine.

By 'hitting aws', i just meant that these are ultimately running on aws, and the opta files create resources on AWS. would it still make sense to send metrics to gcp?

Let's table this in any case.  we don't really have the capacity right now to watch and maintain another cluster.  hopefully by the end of the year we're more set up for that.

Reply:
Metrics:  
* AWS-vs-GCP seems pretty irrelevant.  Since this can produce metrics that I want to query, I need persistent storage somewhere that supports the querying I anticipate wanting to do.  The data volume is negligible/small, so no data egress concerns.  Is there another serverless datastore on AWS that supports querying that you'd suggest?  Or something else that the OSS Project uses for storing metrics of various kinds that can be searched?  I'm asking for the low-cost, low-maintenance data store where I can publish metrics and query.  That is most easily done by me on BigQuery -- but if something else is being used, then I can use that.  I don't understand not creating something to collect metrics, and can setup something to initially do that for the OSS project, if something doesn't already exist.  



Maintaining another cluster:
* This ultimately will create and destroy a cluster nightly, and on releases.  Therefore at least once a day there will be a new cluster and then it will be destroyed.  In the event wanting to change the code/testing process, it would make sense to do in a separate flow on a different cluster in a different project/account ( only at the time that making changes to a codebase ), therefore rarely be changed/run.   But, I don't think that's something that requires additional capacity from anyone at Union [ something would run only on the time that someone is making a code change to the GH Workflows, and that would just ensure not to mess with prod until PR to change ].  The monitoring method would be the same, so not require additional maintenance burden [ only incredibly small initial setup cost, since I don't have permissions to do myself ].  I'm saying I disagree with your premise, so wanting to ensure you see why.  Though, if you don't want me to create that, then that's fine.  Let's just be explicit about why.

Reply:
sorry, I meant to say account.  We currently do not have a GCP account.  None of the billing/alerting/monitoring to make sure it hasn't been hijacked for bitcoin mining is set up. There's some overhead associated with that and I'd like to push that off til we've a bit more breathing room.

Reply:
I can help with GCP "project" configuration [ projects are the primitive there ], and Flyte does have GCP Account/Organization, per @EngHabu [ that was part of the impetus for getting better testing in place and my involvement ] .  

Billing/monitoring setup is pretty easy [ I did manage multiple GCP Organizations and setup the project factories/etc across multiple organizations in prior roles ], once you're ready [ sounds like not now ] I can show you how.    

Probably will also want to see:  https://github.com/terraform-google-modules/terraform-google-project-factory and related

Reply:
Without a FlyteOSS alternate preference for metrics/logging, I'll create something that works, and we can extend/modify and get into a better system eventually.